rskj-core/src/main/java/co/rsk/net/sync/SyncConfiguration.java

```diff
@@ -89,6 +127,10 @@ public SyncConfiguration(
         // TODO(snap-poc) re-visit the need of this specific timeout as the algorithm evolves
         this.timeoutWaitingSnapChunk = Duration.ofSeconds(timeoutWaitingSnapChunk);
         this.snapshotSyncLimit = snapshotSyncLimit;
+        this.snapshotTrustedPeers = snapshotTrustedPeers != null ? snapshotTrustedPeers : Collections.emptyList();
+
+        nodeIdToSnapshotTrustedPeerMap = getSnapshotTrustedPeers().stream()
+                .collect(Collectors.toMap(peer -> peer.getId().toString(), peer -> peer));
```

Comment:
```suggestion

        nodeIdToSnapshotTrustedPeerMap = Collections.unmodifiableMap(getSnapshotTrustedPeers().stream()
                .collect(Collectors.toMap(peer -> peer.getId().toString(), peer -> peer)));
```

Reply:
could you pls rebase onto the base branch?

Reply:
solved, added this change and rebased the branch.